kinto_client/importer.py

```diff
@@ -0,0 +1,348 @@
+                                help='Bucket name, usually the app name',
+                                type=str, default=self.default_bucket)
+
+            parser.add_argument('-c', '--collection',
+                                help='Collection name',
+                                type=str, default=self.default_collection)
+        if include_authentication:
+            parser.add_argument('-u', '--auth', help='BasicAuth user:pass',
+                                type=str, default=self.default_auth)
+
+        if include_files:
+            nargs = '+'
+            if self.default_files is not None:
+                nargs = '*'
+
+            parser.add_argument('files', metavar='N', type=str, nargs=nargs,
+                                help='A list of files to import.',
+                                default=self.default_files)
+
+        if include_verbosity:
+            parser.add_argument('-v', '--verbose',
+                                help='Display status',
+                                dest='verbose',
+                                action='store_true')
+
+        return parser
+
+    def get_arguments(self, parser, args=None):
+        args = vars(parser.parse_args(args=args))
+
+        files = []
+
+        if 'files' in args:
+            for f in args['files']:
+                if os.path.exists(f):
+                    files.append(os.path.abspath(f))
+                    command_log(self.logger, '%s: ✓' % os.path.abspath(f))
+                else:
+                    self.logger.error('%s: ✗' % os.path.abspath(f))
+
+            args['files'] = files
+
+        auth = args.get('auth')
+
+        if auth:
+            args['auth'] = self.get_auth(auth)
+
+        return args
+
+    def get_auth(self, auth):
+        """Ask for the user password if needed."""
+        auth = tuple(auth.split(':', 1))
+        if len(auth) < 2:
+            email = auth[0]
+            password = getpass.getpass('Please enter a password for %s: '
+                                       % email)
+            auth = (email, password)
+
+        return auth
+
+    def setup_logger(self, logger=None):
+        """Configure the logger with regards to the verbosity param."""
+        logger = logger or self.logger
+        if 'verbose' in self.args and self.args['verbose']:
+            logger.setLevel(logging.INFO)
+
+    def get_local_records(self):
+        self.logger.debug('Reading data from files')
+        raise NotImplementedError()
+
+    def setup_local_client(self, local_client=None):
+        return local_client
+
+    def setup_remote_client(self, remote_client=None):
+        # If the client is already created, just return it.
+        if remote_client:
+            return remote_client
+
+        # If the include_remote_server functionnality is not activated at the
+        # parser level do not create the remote_client.
+        if not self.include_remote_server:
+            return
```

Comment:
Because you may want to pass along the remote_client without actually configuring the parser to allow remote server configuration.